org.eclipse.gef/src/org/eclipse/gef/editparts/AbstractGraphicalEditPart.java

```diff
@@ -803,22 +764,8 @@ public void removeNodeListener(NodeListener listener) {
 	 */
 	@Override
 	public void removeNotify() {
-		List conns;
-		ConnectionEditPart cep;
-		conns = getSourceConnections();
-		for (Object conn : conns) {
-			cep = (ConnectionEditPart) conn;
-			if (cep.getSource() == this) {
-				cep.setSource(null);
-			}
-		}
-		conns = getTargetConnections();
-		for (Object conn : conns) {
-			cep = (ConnectionEditPart) conn;
-			if (cep.getTarget() == this) {
-				cep.setTarget(null);
-			}
-		}
+		getSourceConnections().forEach(conn -> conn.setSource(null));
+		getTargetConnections().forEach(conn -> conn.setTarget(null));
```

Comment:
Is there a reason you no longer check for `conn.getSource()/conn.getTarget() == this` or is that merely an oversight?

Reply:
was a stupid copy paste issue. Thx for noting. Fixed now.

Reply:
That's what the reviews are for ;D